### PR TITLE
Feat: add json flags

### PR DESCRIPTION
**Pull request by xfeng96**

### CODE REVIEW

File: parse/parse.go [hunk 1]
```diff
@@ -5,10 +5,9 @@ package parse
 
 import (
 	"encoding/json"
-	"fmt"
 	"strings"
 
-	"github.com/waflab/waflab/object"
+	mapset "github.com/deckarep/golang-set"
```
Comment (hsluoyz) [inline parse/parse.go:10]: Why not use map?

File: parse/parse.go [hunk 2]
```diff
@@ -44,22 +43,16 @@ type WafRule struct {
...
-	testcases := object.GetFilteredTestcases("autogen-test")
-	testcaseMap := map[string]*object.Testcase{}
-	for _, testcase := range testcases {
-		name := strings.TrimLeft(testcase.Name, "autogen-")
-		name = strings.TrimRight(name, ".yaml")
-		testcaseMap[name] = testcase
-	}
-
-	s := util.ReadStringFromPath(util.EnabledRulePath)
+func GetEnabledRules(path string) mapset.Set {
```
Comment (hsluoyz) [inline parse/parse.go:46]: Don't replace original code. Create a new .go file to hold the new code

File: cmd/README.md [hunk 3]
```diff
@@ -170,4 +170,13 @@ Change the entry separator
...
 
 944250-7,403,920320 944250 949110  
+```
+
+#### Json
+
+Json flags allows you to read enabled rules in a specified json file and only these rules will be tested.
+
+```bash
+waflab test [target WAF address] --json [JSON path]
```
Comment (hsluoyz) [inline cmd/README.md:180]: We should support a default value for --json. In an internal branch, we will ship the default JSON file.

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: parse/parse.go (hunk 1 above)
Comment (Olament) [reply parse/parse.go:10]: You can use ```map[type]bool``` as a set in Golang

Comment (xfeng96) [reply parse/parse.go:10]: > You can use `map[type]bool` as a set in Golang

got it

Comment (xfeng96) [reply parse/parse.go:10]: > You can use `map[type]bool` as a set in Golang

Got it.

File: cmd/command/test.go [hunk 4]
```diff
@@ -79,7 +101,18 @@ func testing(cmd *cobra.Command, args []string) {
 				if err != nil {
 					return err
 				}
-				yamlTestcases = append(yamlTestcases, string(out))
+				testTitle := path[11:]
```
Comment (Olament) [inline cmd/command/test.go:104]: What is the magic number 11 for? If you are trying to match the test title, use following regular expression instead. 

```go
re := regexp.MustCompile(`(\d+)\.yaml`)
strs := re.FindStringSubmatch(`123213213/autogen-930120.yaml`)
if len(strs) > 1 {
        // obtain test title succesfully
	// accessing testtile using strs[1]
} else {
	// failed to obtain test title
}
```

**Comment by Olament** [comment]
Looks good to me now. @hsluoyz

File: cmd/command/test.go [hunk 5]
```diff
@@ -47,6 +50,13 @@ func testing(cmd *cobra.Command, args []string) {
 	}
 
 	var yamlTestcases []string
+
+	// get enabled rules map from wafrules.json or wafrules-drs-2.0.json
+	var enabledRuleSet map[string]bool
+	if _, err := os.Stat(jsonPath); !os.IsNotExist(err) {
```
Comment (hsluoyz) [inline cmd/command/test.go:56]: Use: `util.FileExist()`

File: cmd/command/test.go [hunk 6]
```diff
@@ -70,7 +83,16 @@ func testing(cmd *cobra.Command, args []string) {
 				fmt.Fprintln(os.Stderr, err)
 				return
 			}
-			yamlTestcases = append(yamlTestcases, string(out))
+			if _, err := os.Stat(jsonPath); !os.IsNotExist(err) {
```
Comment (hsluoyz) [inline cmd/command/test.go:86]: Why check file path again? You can just check whether enabledRuleSet is nil.

File: cmd/command/test.go [hunk 7]
```diff
@@ -79,7 +101,23 @@ func testing(cmd *cobra.Command, args []string) {
 				if err != nil {
 					return err
 				}
-				yamlTestcases = append(yamlTestcases, string(out))
+				re := regexp.MustCompile(`(\d+)\.yaml`)
+				strs := re.FindStringSubmatch(path)
+				if len(strs) > 1 {
+					testTitle := strs[1]
+					if _, err := os.Stat(jsonPath); !os.IsNotExist(err) {
```
Comment (hsluoyz) [inline cmd/command/test.go:108]: Large section of duplicated code. Can you minimize the if-else branch part and put it in the front?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**